pkg/adaptor/hypervisor/aws/ec2.go

```diff
@@ -76,3 +76,42 @@ func NewEC2Client(cloudCfg Config) (*ec2.Client, error) {
 	client := ec2.NewFromConfig(cfg)
 	return client, nil
 }
+
+func RetrieveMissingConfig(cfg *Config) error {
+	mdr := NewMetadataRetriever()
```

Comment:
checking for mdr ?

Reply:
it fails only if it cannot fetch the mac address, which is not always required and if some field does require it will fail on Get
(imds client creating also doesn't seem to fail)

Reply:
ok. May be add a comment just above this call for future use.